src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -452,6 +471,76 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
+      .getAffectedIndividualDeadlines(submissionEndTimestamp);
+    const [affectedStudentModels, affectedInstructorModels] = this
+      .getAffectedIndividualModels(submissionEndTimestamp, affectedStudentDeadlines, affectedInstructorDeadlines);
+
+    const modalRef: NgbModalRef = this.ngbModal.open(ExtensionConfirmModalComponent);
+    modalRef.componentInstance.modalType = ExtensionModalType.VALIDATE_DEADLINE;
+    modalRef.componentInstance.selectedStudents = affectedStudentModels;
+    modalRef.componentInstance.selectedInstructors = affectedInstructorModels;
+    modalRef.componentInstance.extensionTimestamp = submissionEndTimestamp;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.sessionEditFormModel.timeZone;
+
+    return new Observable((subscribeIsUserAccept) => {
+      modalRef.componentInstance.onConfirmExtensionCallBack.subscribe(() => {
+        this.updateDeadlines(affectedStudentModels, affectedInstructorModels, affectedStudentDeadlines,
+          affectedInstructorDeadlines);
+        modalRef.componentInstance.isSubmitting = false;
+        modalRef.close();
+        subscribeIsUserAccept.next(true);
+      }, () => {
+        subscribeIsUserAccept.next(false);
+      });
+    });
+  }
+
+  private getAffectedIndividualDeadlines(submissionEndTimestamp: number): [
+    Record<string, number>, Record<string, number>,
+  ] {
+    const affectedStudentDeadlines = DeadlineExtensionHelper.setDeadlinesBeforeEndTime(
+      this.studentDeadlines, submissionEndTimestamp);
+    const affectedInstructorDeadlines = DeadlineExtensionHelper.setDeadlinesBeforeEndTime(
+      this.instructorDeadlines, submissionEndTimestamp);
+    return [affectedStudentDeadlines, affectedInstructorDeadlines];
+  }
+
+  private getAffectedIndividualModels(
+    submissionEndTimestamp: number,
+    affectedStudentDeadlines: Record<string, number>,
+    affectedInstructorDeadlines: Record<string, number>,
+  ): [StudentExtensionTableColumnModel[], InstructorExtensionTableColumnModel[]] {
+    const affectedStudents = this.studentsOfCourse.filter((student) => affectedStudentDeadlines[student.email]);
+    const affectedInstructors = this.instructorsCanBePreviewedAs
+      .filter((instructor) => affectedInstructorDeadlines[instructor.email]);
+
+    const affectedStudentModels = DeadlineExtensionHelper.mapStudentsToStudentModels(
+      affectedStudents, affectedStudentDeadlines, submissionEndTimestamp,
+    );
+    const affectedInstructorModels = DeadlineExtensionHelper.mapInstructorsToInstructorModels(
+      affectedInstructors, affectedInstructorDeadlines, submissionEndTimestamp,
+    );
+    return [affectedStudentModels, affectedInstructorModels];
+  }
+
+  updateDeadlines(affectedStudents: StudentExtensionTableColumnModel[],
+    affectedInstructors: InstructorExtensionTableColumnModel[], affectedStudentDeadlines: Record<string, number>,
+    affectedInstructorDeadlines: Record<string, number>,
+  ): void {
```

Comment:
Small suggestion about naming:

```suggestion
  private deleteDeadlines(students: StudentExtensionTableColumnModel[],
    instructors: InstructorExtensionTableColumnModel[], studentDeadlines: Record<string, number>,
    instructorDeadlines: Record<string, number>,
  ): void {
```

Rationale:
1. I believe this can be private unless i'm mistaken.
2. `deleteDeadlines` might be more representative of the action being done here.
3. Removing "affected" from the param names: Even though it is only used in a very specific case to delete deadlines for "affected" users, the method itself is quite generic and shouldn't care about whether the users are "affected" or not.

Reply:
I was initially hesitant to use `deleteDeadlines`, as I realized there's no actual deletion / CRUD to the database. Now I realize `updateDeadlines` can also be misinterpreted. Would `setDeadlines` and some comments be better?

Reply:
How about `removeDeadlines` then? I don't really see an issue, since you are removing deadlines from the object. Being pedantic but but seems odd to me to use update and set when we really mean "remove all these deadlines that we pass in".